__tests__/integration/atomic-swap.test.js

```diff
@@ -0,0 +1,726 @@
+
+    data = response.body.data;
+
+    // wallet1: sign data
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal/get-my-signatures')
+      .send({ partial_tx: data })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    loggers.test.insertLineToLog('atomic-swap[1TK P2PKH]: get signatures1', { body: response.body });
+    expect(response.body).toEqual({
+      success: true,
+      signatures: expect.any(String),
+    });
+    const signatureWallet1 = response.body.signatures;
+
+    // wallet2: sign data
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal/get-my-signatures')
+      .send({ partial_tx: data })
+      .set({ 'x-wallet-id': wallet2.walletId });
+    loggers.test.insertLineToLog('atomic-swap[1TK P2PKH]: get signatures2', { body: response.body });
+    expect(response.body).toEqual({
+      success: true,
+      signatures: expect.any(String),
+    });
+    const signatureWallet2 = response.body.signatures;
+
+    // wallet1: build and push data
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal/sign-and-push')
+      .send({ partial_tx: data, signatures: [signatureWallet1, signatureWallet2] })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    loggers.test.insertLineToLog('atomic-swap[1TK P2PKH]: push', { body: response.body });
+    expect(response.body.success).toBe(true);
+
+    await TestUtils.pauseForWsUpdate();
+
+    // Get the balance state after the swap
+    const finalBalance1 = await wallet1.getBalance(tokenTx1.hash);
+    const finalBalance2 = await wallet2.getBalance(tokenTx1.hash);
+
+    // Expect both balances to be changed by 10
+    expect(finalBalance1.available - startBalance1.available).toEqual(-10);
+    expect(finalBalance2.available - startBalance2.available).toEqual(10);
+  });
+
+  it('should exchange multiple tokens between 2 wallets', async () => {
+    // Get the balance state before the swap
+    // balances for wallet1 (HTR, token 1 and token 2)
+    const startHTRBalance1 = await wallet1.getBalance();
+    const startTK1Balance1 = await wallet1.getBalance(tokenTx1.hash);
+    const startTK2Balance1 = await wallet1.getBalance(tokenTx2.hash);
+
+    // balances for wallet2 (HTR, token 1 and token 2)
+    const startHTRBalance2 = await wallet2.getBalance();
+    const startTK1Balance2 = await wallet2.getBalance(tokenTx1.hash);
+    const startTK2Balance2 = await wallet2.getBalance(tokenTx2.hash);
+
+    // wallet1 proposes the transaction
+    // wallet1 will:
+    //  - send 5 TKW1
+    //  - send 10 HTR
+    //  - receive 15 TKW2
+    let response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({
+        send: {
+          tokens: [
+            { token: tokenTx1.hash, value: 5 },
+            { token: '00', value: 10 },
+          ],
+        },
+        receive: {
+          tokens: [{ token: tokenTx2.hash, value: 15 }],
```

Comment:
aI think we should have at least one test choosing the address and validating it was really sent to it.

Reply:
Created a test for specifying addresses on the atomic-swap (both for send change and receive)